app/src/containers/CartPage.tsx

```diff
@@ -179,28 +248,48 @@ class CartPage extends React.Component<RouteComponentProps, CartPageState> {
     return ('');
   }
 
+  handleModalOpen() {
+    this.setState({
+      openModal: true,
+    });
+  }
+
+  handleModalClose() {
+    this.setState({
+      openModal: false,
+    });
+  }
+
+  handleCartSelect(cartData) {
+    this.setState({ cartData });
+  }
+
+  handleCartElementSelect(index) {
+    const { cartsData } = this.state;
+    this.setState({ cartData: cartsData._element[index] });
+  }
+
   render() {
-    const { cartData, isLoading } = this.state;
+    const {
+      cartData, cartsData, isLoading, openModal, multiCartsAvailable,
+    } = this.state;
     const itemDetailLink = '/itemdetail';
+    const cartName = cartData && cartData._descriptor[0].default !== 'true' ? cartData._descriptor[0].name : intl.get('default');
```

Comment:
This will throw an error if multi-cart is not available. Need to validate if the descriptor is present first in this
![image](https://user-images.githubusercontent.com/39800563/66931129-a37da780-f003-11e9-8c78-7957b56ccf82.png)